File: tool/python/data_utils.py
import glob
import pandas as pd


def read_year_data(year):
    file = pd.read_csv('./dados/ocorrenciasmun-brasil{}.csv'.format(year), encoding='latin-1',sep=';',skiprows=4)
    return format_columns(file)

def format_columns(data):
    data.columns=[
    "regiao",
    "uf",
    "estado",
    "codigo_IBGE",
    "municipio",
    "crime",
    "ano",
    "mes",
    "ocorrencias"
    ]
    return data


def read_all_data():
    '''Esta função carrega todos os dados dos csvs de ocorrencias do Brasil
    '''
    files = glob.glob('dados/*.csv')
    years = []
    for file in files:
        data = pd.read_csv(file, encoding='latin-1',sep=';',skiprows=4)
        years.append(data)
    return format_columns(pd.concat(years))

def format_legend(legend_handless):
    legend = []
    for label in legend_handless[1]:
            legend.append(label.split(",")[1][:-1])
    return legend

def create_month_xlegend(months):
    xlegend = ['']
    for month in months:
        if(int(month[1].split('/')[0]) % 2 != 0):
            xlegend.append(month[1])
    return xlegend

def create_year_xlegend(years):
    xlegend = ['']
    for year in years:
        if(int(year) % 2 == 0):
            xlegend.append(year)
    return xlegend

File: tool/r/data_utils.R
'
Funções para a leitura, limpeza e organização dos dados utilizados
'

#' Função para tratar as colunas dos dados (Remove a coluna de código do município) e
#' renomeia as colunas para ficar mais simples a manipulação 
#'
transform.column <- function(data) {
  # data$`Código.IBGE.Município` <- NULL;
  colnames(data) <- c('regiao', 'sigla_uf', 'uf', 'cod_uf', 'municipio', 
                         'tipo_crime', 'mes', 'mes_ano', 'ocorrencias');  
  return(data);
}

#' Função para a leitura de todos os dados do diretório `data`
#' @return data.frame
readAllData <- function(){
  path <- './dados/';
  files <- dir(path);
  allData <- data.frame();
  
  for(f in files) {
    if (!grepl('.pdf', f)) {
      print(f);
      data <- read.csv(paste(path, f, sep = ''), skip = 4, sep = ';', fileEncoding = 'latin1');
      allData <- rbind(allData, data);
    }
  }
  return(transform.column(allData));
}

#' Função para a leitura de um ano específico de ocorrências
#' @return data.frame
readSpecificYear <- function(year) {
  path <- './dados/';
  files <- dir(path);
  
  for (f in files) {
    if (grepl(toString(year), f)) {
      data <- read.csv(paste(path, f, sep = ''), skip = 4, sep = ';', fileEncoding = 'latin1');
      return(transform.column(data));
    } 
  }
}
